hack/packages/templates/repo-utils/package-cr-overlay.yaml

```diff
@@ -2,20 +2,36 @@
 #@ load("@ytt:overlay", "overlay")
 #@ load("package-helpers.lib.yaml", "get_package_repository", "get_package", "get_package_spec")
 
-#@ package_repository = get_package_repository(data.values.packageRepository)
+#@ package_repository = get_package_repository(data.values.packageRepository, data.values.packageName)
 #@ package = get_package(package_repository, data.values.packageName)
 #@ packageSpec = get_package_spec(package_repository, package)
 
 #@ if not hasattr(package, "packageSubVersion") or package.packageSubVersion == "":
-#@ packageVersion = package.version
+#@  if package.version == "latest":
+#@    packageVersion = data.values.version
+#@  else:
+#@    packageVersion = package.version
+#@  end
 #@ else:
-#@ packageVersion = package.version + "+" + package.packageSubVersion
+#@  if package.version == "latest":
+#@    packageVersion = data.values.version + "+" + package.packageSubVersion
+#@  else:
+#@    packageVersion = package.version + "+" + package.packageSubVersion
+#@  end
 #@ end
 
 #@ if not hasattr(package, "packageSubVersion") or package.packageSubVersion == "":
-#@ imagePackageVersion = "v" + package.version
+#@  if package.version == "latest":
+#@    imagePackageVersion = "v" + data.values.version
+#@  else:
+#@    imagePackageVersion = "v" + package.version
+#@  end
 #@ else:
-#@ imagePackageVersion = "v" + package.version + "_" + package.packageSubVersion
+#@  if package.version == "latest":
+#@    imagePackageVersion = "v" + data.values.version + "_" + package.packageSubVersion
+#@  else:
+#@    imagePackageVersion = "v" + package.version + "_" + package.packageSubVersion
+#@  end
```

Comment:
Question: Do we have any docs that I could educate myself on the naming conventions for versions ? I see some have "+" and some are using "_"

Reply:
I don't think we have docs on this and I'm not sure the convention is entirely clear to me either (some of it is derived from downstream). If this non-blocking, we can take this as an action item and include docs on this later.

Reply:
I would suggest opening an issue @danniel1205 or we can discuss what sort of naming convention documentation we should have in one of our team meetings. One of us can open an issue from there.

Reply:
Here is what @yharish991 said,

> OCI specification doesn't support full semver specification, so we need to add _ for images and kapp-controller supports full semver, so we need to use + if we need to add any build metadata to the Package CR version
> Some links for context:
> https://semver.org/#spec-item-10
> https://github.com/opencontainers/distribution-spec/issues/154